agent/command/git.go

```diff
@@ -70,7 +70,8 @@ type gitFetchProject struct {
 	// Note: If a module does not have a revision it will use the module's branch to get the project.
 	Revisions map[string]string `plugin:"expand"`
 
-	Token string `plugin:"expand" mapstructure:"token"`
+	Token   string `plugin:"expand" mapstructure:"token"`
+	isOauth bool   `mapstructure:"is_oauth"`
```

Comment:
what does the plugin and mapstructure exactly do?

Reply:
plugin is used for expansions, mapstructure for structs

Reply:
is there a reason we wouldnt want users to use an expansion to fill this out?

Reply:
because then it would need to be a string. I don't expect users to ever need to case on it so I don't think an expansion is necessary.